rosetta-source/src/main/rosetta/product-qualification-func.rosetta

```diff
@@ -681,6 +687,54 @@ func Qualify_BaseProduct_EquityForward: <"Qualifies a product as having the Asse
         Qualify_AssetClass_Equity(economicTerms) = True
             and economicTerms -> payout -> forwardPayout only exists
 
+func Qualify_EquityForward_PriceReturnBasicPerformance_SingleName: <"Qualifies a product as a plain vanilla Equity Option.  The determination of the qualification is based on the economic terms and the following criteria: 1) An option product  for which the underlier is a single stock and 2) No special option feature exists other than option averaging."> //
+    [qualification Product]
+    inputs:
+        economicTerms EconomicTerms (1..1)
+    output:
+        is_product boolean (1..1)
+            [synonym ISDA_Taxonomy_v1 value "EquityForward_PriceReturnBasicPerformance_SingleName"]
+            [synonym ISDA_Taxonomy_v2 value "EquityForward_PriceReturnBasicPerformance_SingleName"]
+    set is_product:
+        Qualify_AssetClass_Equity(economicTerms) = True
+                // qualifies that only the option payout exist and all other payouts are absent
+            and economicTerms -> payout -> forwardPayout only exists
+                // qualifies that the underlier is an equity security
+            and economicTerms -> payout -> forwardPayout only-element -> underlier -> security exists
+                // qualifies that an option style is defined and that no feature other than averaging exists
+ 
+func Qualify_EquityForward_PriceReturnBasicPerformance_SingleIndex: <"Qualifies a product as a plain vanilla Equity Option.  The determination of the qualification is based on the economic terms and the following criteria: 1) An option product  for which the underlier is a single stock and 2) No special option feature exists other than option averaging."> //
+    [qualification Product]
+    inputs:
+        economicTerms EconomicTerms (1..1)
+    output:
+        is_product boolean (1..1)
+            [synonym ISDA_Taxonomy_v1 value "EquityForward_PriceReturnBasicPerformance_SingleIndex"]
+            [synonym ISDA_Taxonomy_v2 value "EquityForward_PriceReturnBasicPerformance_SingleIndex"]
+    set is_product:
+        Qualify_AssetClass_Equity(economicTerms) = True
+                // qualifies that only the option payout exist and all other payouts are absent
+            and economicTerms -> payout -> forwardPayout only exists
+                // qualifies that the underlier is an equity security
```

Comment:
isn't this qualifying an index rather than an equity?

Reply:
Looks OK to me - logic is. It qualifies when all of the below are true:

1. Qualify_AssetClass_Equity (its an equity)
2. has 1 `forwardPayout`
3. the `forwardPayout` has an `underlier` that is an `index`